### PR TITLE
docs(service-worker): improve service worker guides

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
improve slightly the angular.io guides related to service workers, by
means of fixes, styling, information and clarity changes

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

@aikidave can I count on you on this PR? :heart_eyes: :bow: :pray:

### CODE REVIEW

File: aio/content/examples/service-worker-getting-started/src/app/check-for-update.service.ts [hunk 1]
```diff
@@ -7,8 +7,9 @@ import { first } from 'rxjs/operators';
 export class CheckForUpdateService {
 
   constructor(appRef: ApplicationRef, updates: SwUpdate) {
-    // Allow the app to stabilize first, before starting polling for updates with `interval()`.
+    // Allow the app to stabilize first
```
Comment (dario-piotrowicz) [inline aio/content/examples/service-worker-getting-started/src/app/check-for-update.service.ts:10]: This change is mainly because I wanted to avoid the ugly line-break
Screenshot at 2021-09-18 21-38-24
:

File: aio/content/examples/service-worker-getting-started/src/app/handle-unrecoverable-state.service.ts [hunk 2]
```diff
@@ -9,8 +9,10 @@ export class HandleUnrecoverableStateService {
   constructor(updates: SwUpdate) {
     updates.unrecoverable.subscribe(event => {
       notifyUser(
-        `An error occurred that we cannot recover from:\n${event.reason}\n\n` +
-        'Please reload the page.');
+        'An error occurred that we cannot recover from:\n' +
+        event.reason +
+        '\n\nPlease reload the page.'
```
Comment (dario-piotrowicz) [inline aio/content/examples/service-worker-getting-started/src/app/handle-unrecoverable-state.service.ts:14]: since we are already relying on string concatenation with `+` I figured we could just avoid mixing string interpolation in too

File: aio/content/guide/app-shell.md [hunk 3]
```diff
@@ -87,4 +87,4 @@ Or to use the production configuration.
 ng run my-app:app-shell:production
 </code-example>
 
-To verify the build output, open `dist/my-app/browser/index.html`. Look for default text `app-shell works!` to show that the application shell route was rendered as part of the output.
+To verify the build output, open dist/my-app/bro<span>wser</span>/index.html</code>. Look for default text `app-shell works!` to show that the application shell route was rendered as part of the output.
```
Comment (dario-piotrowicz) [inline aio/content/guide/app-shell.md:90]: This is works fine, but there must most likely be a better solution here, what I needed to do was so break the automatic link generated for the browser which is just plain wrong (and redirects to the totally unrelated animations api)

Screenshot at 2021-09-18 21-40-22

File: aio/content/guide/service-worker-config.md [hunk 4]
```diff
@@ -240,9 +240,9 @@ To use this strategy set `strategy` to `freshness` and `timeout` to `0u` in `cac
 This will essentially do the following:
 
 1. Try to fetch from the network first.
-2. If the network request does not complete after 0ms (that is, immediately), fall back to the cache (ignoring cache age).
-3. Once the network request completes, update the cache for future requests.
-4. If the resource does not exist in the cache, wait for the network request anyway.
+1. If the network request does not complete after 0ms (that is, immediately), fall back to the cache (ignoring cache age).
+1. Once the network request completes, update the cache for future requests.
+1. If the resource does not exist in the cache, wait for the network request anyway.
```
Comment (dario-piotrowicz) [inline aio/content/guide/service-worker-config.md:245]: just using `1.`s to be consistent with other lists I have seen in aio

the same applied to the other lists below

File: aio/content/guide/service-worker-devops.md [hunk 5]
```diff
@@ -23,7 +23,7 @@ This file integrity is especially important when lazy loading modules.
 A JS bundle may reference many lazy chunks, and the filenames of the
 lazy chunks are unique to the particular build of the application. If a running
 application at version `X` attempts to load a lazy chunk, but the server has
-updated to version `X + 1` already, the lazy loading operation will fail.
+already updated to version `X + 1`, the lazy loading operation will fail.
```
Comment (dario-piotrowicz) [inline aio/content/guide/service-worker-devops.md:26]: this seems just easier to read to me :slightly_smiling_face:

File: aio/content/guide/service-worker-getting-started.md [hunk 6]
```diff
@@ -68,14 +72,16 @@ If you are not using HTTPS, the service worker will only be registered when acce
...
-To simulate a network issue, disable network interaction for your application. In Chrome:
+To simulate a network issue, disable network interaction for your application.
+
+In Chrome:
 
 1. Select **Tools** > **Developer Tools** (from the Chrome menu located at the top right corner).
-2. Go to the **Network tab**.
-3. Check the **Offline box**.
+1. Go to the **Network tab**.
+1. Select **offline** from the throttling selection.
```
Comment (dario-piotrowicz) [inline aio/content/guide/service-worker-getting-started.md:81]: This has slightly changed in Chrome so I wanted to keep it up to date :slightly_smiling_face:

File: aio/content/guide/service-worker-getting-started.md [hunk 7]
```diff
@@ -90,7 +96,7 @@ If you look at the Network tab, you can verify that the service worker is active
   <img src="generated/images/guide/service-worker/sw-active.png" alt="Requests are marked as from ServiceWorker">
 </div>
 
-Notice that under the "Size" column, the requests state is `(from ServiceWorker)`. This means that the resources are not being loaded from the network. Instead, they are being loaded from the service worker's cache.
+Notice that under the "Size" column, the requests state is `(ServiceWorker)`. This means that the resources are not being loaded from the network. Instead, they are being loaded from the service worker's cache.
```
Comment (dario-piotrowicz) [inline aio/content/guide/service-worker-getting-started.md:99]: Chrome does't display the `from` anymore

File: aio/content/guide/service-worker-devops.md [hunk 8]
```diff
@@ -96,7 +96,7 @@ configured lifetimes.
 
 It can be problematic for an application if the version of resources
 it's receiving changes suddenly or without warning. See the
-[Versions](guide/service-worker-devops#versions) section above
+[App Versions](guide/service-worker-devops#versions) section above
```
Comment (dario-piotrowicz) [inline aio/content/guide/service-worker-devops.md:99]: For consistency with what we have on line 18:
https://github.com/angular/angular/blob/91183eb14228cf92cb66905a972aca919e6bbaec/aio/content/guide/service-worker-devops.md#L18

(The link is `App Tabs` not just `Tabs`)

File: aio/content/guide/service-worker-devops.md [hunk 9]
```diff
@@ -96,7 +96,7 @@ configured lifetimes.
 
 It can be problematic for an application if the version of resources
 it's receiving changes suddenly or without warning. See the
-[Versions](guide/service-worker-devops#versions) section above
+[App versions](guide/service-worker-devops#versions) section above
```
Comment (dario-piotrowicz) [inline aio/content/guide/service-worker-devops.md:99]: For consistency with what we have on line 18:
https://github.com/angular/angular/blob/91183eb14228cf92cb66905a972aca919e6bbaec/aio/content/guide/service-worker-devops.md#L18

(The link is `App tabs` not `Tabs`)

**Comment by dario-piotrowicz** [comment]
Just wanted to mention that the images I changed are screenshot taken from the latest version of Chrome as the old ones seem to be a bit outdated

I can see that the resolution of my images is higher and they are thus bigger in size.... if that's a problem please let me know and I will reduce their resolution/size

File: aio/content/guide/service-worker-getting-started.md [hunk 10]
```diff
@@ -159,4 +165,5 @@ The service worker installed the updated version of your application *in the bac
 ## More on Angular service workers
 
 You may also be interested in the following:
+* [App Shell](guide/app-shell).
```
Comment (dario-piotrowicz) [inline aio/content/guide/service-worker-getting-started.md:168]: Not totally necessary but since this is next in the nav on the left I figured it could be referenced here (otherwise it's only being referenced in the first introductory page)

File: aio/content/examples/service-worker-getting-started/src/app/check-for-update.service.ts [hunk 11]
```diff
@@ -7,8 +7,9 @@ import { first } from 'rxjs/operators';
...
     const appIsStable$ = appRef.isStable.pipe(first(isStable => isStable === true));
+    // Then poll for updates with `interval()`.
```
Comment (gkalpak) [inline aio/content/examples/service-worker-getting-started/src/app/check-for-update.service.ts:12]: Moving the comment here is a little misleading imo, because the operations do not happen on the lines where the comments are. The comment was supposed to refer to this whole block.

So, could you just restore the original comment and manually wrap it to avoid the line overflow in the code-snippet?

File: aio/content/examples/service-worker-getting-started/src/app/handle-unrecoverable-state.service.ts (hunk 2 above)
Comment (gkalpak) [reply aio/content/examples/service-worker-getting-started/src/app/handle-unrecoverable-state.service.ts:14]: 😞

File: aio/content/guide/app-shell.md (hunk 3 above)
Comment (gkalpak) [inline aio/content/guide/app-shell.md:90]: Good catch!

You can use the [no-auto-link][1] class:
```suggestion
To verify the build output, open <code class="no-auto-link">dist/my-app/browser/index.html</code>. Look for default text `app-shell works!` to show that the application shell route was rendered as part of the output.
```

[1]: https://angular.io/guide/docs-style-guide#auto-linking-in-code-snippets

File: aio/content/guide/service-worker-communications.md [hunk 12]
```diff
@@ -87,7 +87,7 @@ For example, imagine the following scenario:
 In the above scenario, the service worker is not able to serve an asset that would normally be cached.
 That particular application version is broken and there is no way to fix the state of the client without reloading the page.
 In such cases, the service worker notifies the client by sending an `UnrecoverableStateEvent` event.
-You can subscribe to `SwUpdate#unrecoverable` to be notified and handle these errors.
+You can subscribe to `SwUpdate.unrecoverable` to be notified and handle these errors.
```
Comment (gkalpak) [inline aio/content/guide/service-worker-communications.md:90]: OOC, why this change?

File: aio/content/guide/service-worker-config.md [hunk 13]
```diff
@@ -128,7 +128,7 @@ This section describes the resources to cache, broken up into the following grou
 
 ### `cacheQueryOptions`
 
-These options are used to modify the matching behavior of requests. They are passed to the browsers `Cache#match` function. See [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Cache/match) for details. Currently, only the following options are supported:
+These options are used to modify the matching behavior of requests. They are passed to the browsers `Cache.match` function. See [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Cache/match) for details. Currently, only the following options are supported:
```
Comment (gkalpak) [inline aio/content/guide/service-worker-config.md:131]: I don't like this :grin:
`#` indicates that it is an instance method (and not a static method on the `Cache` class).

File: aio/content/guide/service-worker-getting-started.md [hunk 14]
```diff
@@ -55,10 +55,14 @@ http-server -p 8080 -c-1 dist/<project-name>
 
 ### Initial load
 
-With the server running, you can point your browser at http://localhost:8080/. Your application should load normally.
+With the server running, you can point your browser at `http://localhost:8080`. Your application should load normally.
```
Comment (gkalpak) [inline aio/content/guide/service-worker-getting-started.md:58]: ...oooor we could make it a link 🤯 

```suggestion
With the server running, you can point your browser at [http://localhost:8080](http://localhost:8080). Your application should load normally.
```

File: aio/content/guide/service-worker-getting-started.md (hunk 6 above)
Comment (gkalpak) [reply aio/content/guide/service-worker-getting-started.md:81]: Maybe use "dropdown menu" instead of "selection"?

```suggestion
1. Go to the **Network tab**.
1. Select **Offline** in the **Throttling** dropdown menu.
```

File: aio/content/guide/service-worker-getting-started.md [hunk 15]
```diff
@@ -138,7 +144,7 @@ http-server -p 8080 -c-1 dist/<project-name>
 
 Now look at how the browser and service worker handle the updated application.
 
-1. Open http://localhost:8080 again in the same window. What happens?
+1. Open `http://localhost:8080` again in the same window. What happens?
```
Comment (gkalpak) [inline aio/content/guide/service-worker-getting-started.md:147]: Same here: Make it a link?

File: aio/content/guide/service-worker-intro.md [hunk 16]
```diff
@@ -17,25 +17,25 @@ Angular applications, as single-page applications, are in a prime position to be
 
 Angular's service worker is designed to optimize the end user experience of using an application over a slow or unreliable network connection, while also minimizing the risks of serving outdated content.
 
-The Angular service worker's behavior follows that design goal:
+The Angular service worker's behavior follows these design goals:
```
Comment (gkalpak) [inline aio/content/guide/service-worker-intro.md:20]: The points below are not design goals :grin:
They describe the behavior (which follows the design goal mentioned above).

File: aio/content/guide/service-worker-intro.md [hunk 17]
```diff
@@ -17,25 +17,25 @@ Angular applications, as single-page applications, are in a prime position to be
...
 * The service worker conserves bandwidth when possible. Resources are only downloaded if they've changed.
 
-To support these behaviors, the Angular service worker loads a *manifest* file from the server. The manifest describes the resources to cache and includes hashes of every file's contents. When an update to the application is deployed, the contents of the manifest change, informing the service worker that a new version of the application should be downloaded and cached. This manifest is generated from a CLI-generated configuration file called `ngsw-config.json`.
+To support these behaviors, the Angular service worker loads a *manifest* file from the server. The file, called the `ngsw.json` (not to be confused with the `manifest.webmanifest` file), describes the resources to cache and includes hashes of every file's contents. When an update to the application is deployed, the contents of the manifest change, informing the service worker that a new version of the application should be downloaded and cached. This manifest is generated from a CLI-generated configuration file called `ngsw-config.json`.
```
Comment (gkalpak) [inline aio/content/guide/service-worker-intro.md:28]: Good call :100:
Maybe call the other file "web app manifest" and link to [MDN](https://developer.mozilla.org/en-US/docs/Web/Manifest):

```suggestion
To support these behaviors, the Angular service worker loads a *manifest* file from the server. The file, called `ngsw.json` (not to be confused with the [web app manifest](https://developer.mozilla.org/en-US/docs/Web/Manifest)), describes the resources to cache and includes hashes of every file's contents. When an update to the application is deployed, the contents of the manifest change, informing the service worker that a new version of the application should be downloaded and cached. This manifest is generated from a CLI-generated configuration file called `ngsw-config.json`.
```

File: aio/content/guide/service-worker-getting-started.md [hunk 18]
```diff
@@ -148,7 +154,7 @@ What went wrong? Nothing, actually. The Angular service worker is doing its job
 
 If you look at the `http-server` logs, you can see the service worker requesting `/ngsw.json`. This is how the service worker checks for updates.
 
-2. Refresh the page.
+1. Refresh the page.
```
Comment (gkalpak) [inline aio/content/guide/service-worker-getting-started.md:157]: Due to incorrect indentation of the lines above (and below), this is not picked up a second item to the previous list, but as a first item of a new list instead.

Either change it back to `2` or indent the other lines as needed to make the list markup valid (so it is picked up correctly by the Markdown parser).

**Review by gkalpak — CHANGES_REQUESTED** [review]
Nice set of changes 🚀
Thx, @dario-piotrowicz :heart:

File: aio/content/examples/service-worker-getting-started/src/app/check-for-update.service.ts (hunk 11 above)
Comment (dario-piotrowicz) [reply aio/content/examples/service-worker-getting-started/src/app/check-for-update.service.ts]: yes I you're right, I am making things (slightly) more confusing this way :sweat:

File: aio/content/examples/service-worker-getting-started/src/app/handle-unrecoverable-state.service.ts (hunk 2 above)
Comment (dario-piotrowicz) [reply aio/content/examples/service-worker-getting-started/src/app/handle-unrecoverable-state.service.ts:14]: you're ok with this? :sweat: 

I really like string interpolation but it is tricky to get right for multi-lines I think :sweat:

(what I don't like I guess is mixing the two :sweat_smile:)

File: aio/content/guide/app-shell.md (hunk 3 above)
Comment (dario-piotrowicz) [reply aio/content/guide/app-shell.md]: Thanks!
awesome, it's exactly what I needed! :heart_eyes::+1: 

sorry I could not find it by myself :sweat:

File: aio/content/guide/service-worker-communications.md (hunk 12 above)
Comment (dario-piotrowicz) [reply aio/content/guide/service-worker-communications.md:90]: well.... isn't this a typo? :thinking: (am I missing something here? :sweat_smile:)

With it the link is broken/not-fully recognized:
Screenshot at 2021-09-20 21-27-58

(as you can see only the `SwUpdate` bit is recognized)

\
\
With my change the whole link works as I would expect it to:
Screenshot at 2021-09-20 21-27-36

(as you can see it now redirects to the precise property of in the `SwUpdate` docs)

File: aio/content/guide/service-worker-getting-started.md (hunk 6 above)
Comment (dario-piotrowicz) [reply aio/content/guide/service-worker-getting-started.md]: yeah, sorry bad choice of word on my part here :sweat:

File: aio/content/guide/service-worker-getting-started.md (hunk 14 above)
Comment (dario-piotrowicz) [reply aio/content/guide/service-worker-getting-started.md]: mh.... not sure, as this is sort of a broken link unless someone is actually coding-along :man_shrugging: 

but if it's what you prefer it I don't mind making the change :slightly_smiling_face::+1:

File: aio/content/guide/service-worker-intro.md (hunk 16 above)
Comment (dario-piotrowicz) [reply aio/content/guide/service-worker-intro.md:20]: ah ok :sweat_smile::+1: 

Still I don't think the wording then is on point here... :sweat: 

Would you mind if modified it to something like:
```
The Angular service worker follows this design goal by considering the following:
```

or something of that sort? :thinking:

File: aio/content/guide/service-worker-getting-started.md (hunk 18 above)
Comment (dario-piotrowicz) [reply aio/content/guide/service-worker-getting-started.md:157]: sorry! it was such a simple change I didn't even check it!!! :fearful: :grimacing: 

I'm on it! :bow:

Comment (dario-piotrowicz) [reply aio/content/guide/service-worker-getting-started.md:157]: Indentation fixed! :smile: 

it also looks so much cleaner now!!! :heart_eyes::+1: 

Before:
Screenshot at 2021-09-20 22-01-34

After:
Screenshot at 2021-09-20 22-01-52

**Comment by dario-piotrowicz** [comment]
@gkalpak Thanks a lot for the awesome reviewing!!!! (like always! :heart_eyes:)

I addressed all your comments :slightly_smiling_face::+1: 

The only one still not addressed (waiting for your replies) should be:
 - https://github.com/angular/angular/pull/43508#discussion_r712411841
 - https://github.com/angular/angular/pull/43508#discussion_r712402106
 - https://github.com/angular/angular/pull/43508#discussion_r712401446 (maybe? :thinking:)
 - https://github.com/angular/angular/pull/43508#discussion_r712397712 (maybe? :thinking:)

File: aio/content/guide/service-worker-communications.md (hunk 12 above)
Comment (dario-piotrowicz) [reply aio/content/guide/service-worker-communications.md:90]: Ah yeah and just wanted to mention that it is like this in other places too :slightly_smiling_face: 

[Screenshot at 2021-09-20 22-21-47](https://angular.io/guide/service-worker-intro#browser-support)

File: aio/content/guide/service-worker-intro.md (hunk 16 above)
Comment (gkalpak) [reply aio/content/guide/service-worker-intro.md:20]: Fair enough :smiley:
How about:

```suggestion
To achieve this, the Angular service worker follows these guidelines:
```

File: aio/content/guide/service-worker-communications.md (hunk 12 above)
Comment (gkalpak) [reply aio/content/guide/service-worker-communications.md:90]: Yeah, I don't like the `.` for instance members, but our auto-linking tool does not recognize `#` 😞
So, for things that can benefit from auto-linking, `.` makes sense.

File: aio/content/examples/service-worker-getting-started/src/app/handle-unrecoverable-state.service.ts (hunk 2 above)
Comment (gkalpak) [reply aio/content/examples/service-worker-getting-started/src/app/handle-unrecoverable-state.service.ts:14]: I don't like it much, but I'll live 😛

File: aio/content/guide/service-worker-config.md (hunk 13 above)
Comment (gkalpak) [reply aio/content/guide/service-worker-config.md:131]: Well...for other APIs, MDN uses `.prototype.` (for example, [Array.prototype.find()][1])

I really don't like that it is not possible to differentiate between a static member and an instance member with the `.`.
I would normally insist on using `#`, but here things are muddied a bit due to having to use `.` for auto-linking (in other places in the guide), so neither way is ideal.

Leaving it up to you 😛

[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

File: aio/content/guide/service-worker-communications.md (hunk 12 above)
Comment (dario-piotrowicz) [reply aio/content/guide/service-worker-communications.md:90]: well, I must admit my ignorance in that I did not know about `#` being used for instances :sweat_smile:  

I am very used to `.` and I guess I can see your point but in general I think it is sort of clear where the method sits giving giving the context (and even if it's not it becomes immediately clear as soon as you start using it in code :stuck_out_tongue_winking_eye:)

either way yeah here if fixes the linking so it is definitely an improvement I believe :smile:

File: aio/content/guide/service-worker-config.md (hunk 13 above)
Comment (dario-piotrowicz) [reply aio/content/guide/service-worker-config.md:131]: ah yeah, you're totally right about the `.prototype.` :slightly_smiling_face::+1: 
(I guess that would feel terribly out of place in typescript docs :sweat_smile:, given the level of abstraction and all)

anyways I got away with the other `.` in the auto-linking related change so I can change this one to `#` so each of us get one! :smiley:

File: aio/content/guide/service-worker-intro.md (hunk 16 above)
Comment (dario-piotrowicz) [reply aio/content/guide/service-worker-intro.md:20]: yeah! :heart_eyes::+1:

**Review by gkalpak — APPROVED** [review]
Awesome! Thx, @dario-piotrowicz 👍
